svir/dialogs/load_gmf_data_as_layer_dialog.py

```diff
@@ -64,8 +64,40 @@ def __init__(self, iface, viewer_dock, session, hostname, calc_id,
     def set_ok_button(self):
         self.ok_button.setEnabled(self.imt_cbx.currentIndex() != -1)
 
+    def show_num_sites(self):
+        # NOTE: we are assuming all realizations have the same number of sites,
+        #       which is always true for scenario calculations.
+        #       If different realizations have a different number of sites, we
+        #       need to move this block of code inside on_rlz_or_stat_changed()
+        rlz = self.rlz_or_stat_cbx.itemData(
+            self.rlz_or_stat_cbx.currentIndex())
+        gmpe_data = self.npz_file[rlz]
+        self.num_sites_lbl.setText(
+            self.num_sites_msg % gmpe_data.shape)
+
+    def populate_rlz_or_stat_cbx(self):
+        self.rlzs_or_stats = [key for key in sorted(self.npz_file)
+                              if key not in ('imtls', 'array')]
+        self.rlzs_npz = extract_npz(
+            self.session, self.hostname, self.calc_id, 'realizations',
+            message_bar=self.iface.messageBar(), params=None)
+        # NOTE: gmpe and gsim are synonyms
+        self.gmpes = self.rlzs_npz['array']['gsims']
+        assert len(self.rlzs_or_stats) == len(set(self.gmpes)), (
+            "For scenario calculations, the number of realizations should be"
+            " equal to the number of distinct GMPEs.\n"
+            "Realizations: %s\nGMPEs: %s"
+            % (self.rlzs_or_stats, self.gmpes))
```

Comment:
If the user see this, there is nothing she can do. This is actually a bug in the engine. Perhaps we can remove this assert here and add a check in the engine, so that a computation does not even start if we are in this condition.